mlflow/utils/proto_json_utils.py

```diff
@@ -218,7 +218,7 @@ def cast_df_types_according_to_schema(pdf, schema):
         n = min(len(schema.input_types(), pdf.columns))
         dtype_list = zip(pdf.columns[:n], schema.input_types[:n])
 
-    for col_name, col_type_spec in dtype_list:
+    for col_name, col_type_spec in enumerate(dtype_list):
```

Comment:
Is this correct? If `dtype_list` contains tuples, calling `enumerate()` will make it contain triples. Won't the `col_name` / `col_type_spec` being referenced in this loop be wrong?